docs/src/views/IntegrateCustomComponents.tsx

```diff
@@ -0,0 +1,238 @@
+import * as React from 'react'
+import * as cx from 'classnames'
+import { NavLink } from 'react-router-dom'
+import { Header } from 'semantic-ui-react'
+import {
+  Button,
+  Divider,
+  Provider,
+  createComponent,
+  ComponentSlotStyle,
+  ComponentVariablesInput,
+} from '@stardust-ui/react'
+
+import DocPage from '../components/DocPage/DocPage'
+import ExampleSnippet from '../components/ExampleSnippet/ExampleSnippet'
+
+interface StyledButtonProps {
+  className?: string
+  styles?: ComponentSlotStyle
+  variables?: ComponentVariablesInput
+  children?: React.ReactNodeArray | React.ReactNode
+}
+
+const StyledButton: React.SFC<StyledButtonProps> = createComponent<StyledButtonProps, {}>({
```

Comment:
The state is optional.

```suggestion
const StyledButton: React.SFC<StyledButtonProps> = createComponent<StyledButtonProps>({
```